assign-doi-datacite/src/test/java/no/unit/nva/datacite/clients/DataciteClientTest.java

```diff
@@ -0,0 +1,293 @@
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.any;
+import static com.github.tomakehurst.wiremock.client.WireMock.delete;
+import static com.github.tomakehurst.wiremock.client.WireMock.deleteRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.matchingJsonPath;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.put;
+import static com.github.tomakehurst.wiremock.client.WireMock.putRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import com.amazonaws.secretsmanager.caching.SecretCache;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.BasicCredentials;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import java.net.Socket;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.nio.file.Path;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.time.Duration;
+import java.util.Map;
+import java.util.UUID;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.X509ExtendedTrustManager;
+import no.unit.nva.datacite.clients.exception.ClientException;
+import no.unit.nva.datacite.clients.models.Doi;
+import no.unit.nva.datacite.clients.models.ImmutableDoi;
+import no.unit.nva.datacite.config.DataciteConfigurationFactory;
+import no.unit.nva.datacite.config.DataciteConfigurationFactoryForDataciteClientTest;
+import no.unit.nva.datacite.config.PasswordAuthenticationFactory;
+import no.unit.nva.datacite.mdsclient.DataCiteMdsConnection;
+import no.unit.nva.datacite.mdsclient.DataciteMdsConnectionFactory;
+import no.unit.nva.datacite.models.DataCiteMdsClientConfig;
+import no.unit.nva.datacite.models.DataCiteMdsClientSecretConfig;
+import nva.commons.utils.IoUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.apache.http.entity.ContentType;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class DataciteClientTest {
+
+    public static final String EXAMPLE_CUSTOMER_ID = "https://example.net/customer/id/4512";
```

Comment:
Remove unused. Where you can, extract to methods with names that explain what is going on, use constants and use test naming convention.

Reply:
Better now? The code was refactored quite a bit.